src/NuGet.Core/NuGet.Protocol/CachingSourceProvider.cs

```diff
@@ -36,6 +36,32 @@ public CachingSourceProvider(IPackageSourceProvider packageSourceProvider)
                 .ToList();
         }
 
+        public CachingSourceProvider(IEnumerable<SourceRepository> loadedRepositories, IPackageSourceProvider sourceProvider)
```

Comment:
Allows creating a cachingSourceProvider that uses already loaded repositories before creating new repositories from settings.